src/draft/NSG/Agent/LegalEntity/NonListedCompany/Establishment/Write.py

```diff
@@ -0,0 +1,1633 @@
+        description="The nationality of a person",
+        example=ISO_3166_1_Alpha_3.USA,
+    )
+
+
+class BoardMembers(CamelCaseModel):
+    role: BoardMemberRole = Field(
+        ...,
+        title="Role",
+        description="The role of the person in the board",
+        example=BoardMemberRole.MEMBER,
+    )
+    given_name: str = Field(
+        ...,
+        title="Given name",
+        description="The first name that the person is being called by",
+        example="Mary",
+        max_length=250,
+    )
+    middle_names: str = Field(
+        ...,
+        title="Middle names",
+        description="All the middle names of the person",
+        example="Juliet Olive",
+        max_length=250,
+    )
+    last_name: str = Field(
+        ...,
+        title="Last name",
+        description="The person's current family name",
+        example="Deo",
+        max_length=250,
+    )
+    date_of_birth: date = Field(
+        ...,
+        title="Date of birth",
+        description="The birth day of a person",
+        example=date(1976, 4, 16),
+    )
+    nationality: ISO_3166_1_Alpha_3 = Field(
+        ...,
+        title="Nationality",
+        description="The nationality of a person",
+        example=ISO_3166_1_Alpha_3.USA,
+    )
+
+
+class AuditorDetails(CamelCaseModel):
+    company_name: Optional[str] = Field(
+        None,
+        title="Company name",
+        description="The name of the auditor company if exists",
+        example="Auditor company X",
+        max_length=250,
+    )
+    national_identifier: Optional[str] = Field(
+        None,
+        title="National identifier",
+        description="The national identifier of the non-listed company issued by the trade register",
+        example="2464491-9",
+        max_length=40,
+    )
+    given_name: Optional[str] = Field(
+        None,
+        title="Given name",
+        description="The first name that the person is being called by",
+        example="Jane",
+        max_length=250,
+    )
+    last_name: Optional[str] = Field(
+        None,
+        title="Last name",
+        description="The person's current family name",
+        example="Doe",
+        max_length=250,
+    )
+
+
+class EstablishmentRequest(CamelCaseModel):
+    registrant: Registrant = Field(
+        ...,
+        title="Registrant",
+        description="The personal details of person registering the company",
+    )
+    company_details: CompanyDetails = Field(
+        ...,
+        title="Company details",
+        description="The details of the company being established",
+    )
+    share_series: List[ShareSeries] = Field(
+        ...,
+        title="Share series",
+        description="The details of the share series of the company",
+    )
+    company_address: CompanyAddress = Field(
+        ...,
+        title="Company address",
+        description="The official address of the company",
+    )
+    managing_directors: List[ManagingDirectors] = Field(
```

Comment:
I think the class `ManagingDirectors` should be renamed `ManagingDirector`, as each of them describes just one person and here we then use a list of them to get the `managing_directors`.